CMakeLists.txt

```diff
@@ -19,18 +19,30 @@ project(WABT VERSION 1.0.13)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
+# Check if wabt is being used directly or via add_subdirectory, FetchContent, etc
+set(WABT_MASTER_PROJECT OFF)
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(WABT_MASTER_PROJECT ON)
+endif()
+
 # For git users, attempt to generate a more useful version string
 if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
   find_package(Git QUIET REQUIRED)
   execute_process(COMMAND
           "${GIT_EXECUTABLE}" --git-dir=${CMAKE_CURRENT_SOURCE_DIR}/.git describe --tags
           RESULT_VARIABLE
               GIT_VERSION_RESULT
+          ERROR_VARIABLE
+              GIT_VERSION_ERROR
           OUTPUT_VARIABLE
               GIT_VERSION
           OUTPUT_STRIP_TRAILING_WHITESPACE)
   if (${GIT_VERSION_RESULT})
-    message(WARNING "Error running git describe to determine version")
+    # Don't issue warning if we aren't the master project;
+    # just assume that whoever included us knows the version they are getting
+    if (WABT_MASTER_PROJECT)
+      message(WARNING "${GIT_VERSION_ERROR} Error running git describe to determine version")
+    endif()
```

Comment:
Would it be more compatible to actually detect shallow clone?

Reply:
...it didn't occur to me that that was easy to do. Let me see if it is.

Reply:
I'm not sure there is an easy way :)    Just throwing the idea out there.